Support more data types in ONNXIFI transform

yinghai opened the pull request:
Summary: We need to support types beside `int64` and `float`.

Differential Revision: D12967258

rdzhabarov reviewed (approved):
LGTM! +1 for explicit types instead of auto in those cases.

zrphercule commented on caffe2/opt/onnxifi_transformer.cc:
```diff
@@ -89,7 +90,7 @@ std::vector<::ONNX_NAMESPACE::ValueInfoProto> ConvertToValueInfo(
       auto* tensor_type = value_info.mutable_type()->mutable_tensor_type();
       tensor_type->set_elem_type(
-          ::ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT);
+          onnx::Caffe2TypeToOnnxType(it->second.data_type()));
```
Why we are using "onnx" here?

yinghai replied:
caffe2::onnx is a different namespace from ::ONNX_NAMESPACE

Commit:
Support more data types in ONNXIFI transform (#13745)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13745

We need to support types beside `int64` and `float`.

Differential Revision: D12967258

fbshipit-source-id: 8ac68342c2a02328042065922915870b615947aa
